src/blocking/worker.rs

```diff
@@ -30,8 +30,18 @@ where
     BQueue: Queueable + Clone + Sync + Send + 'static,
 {
     pub fn run(&self, task: Task) {
-        let result = self.execute_task(task);
-        self.finalize_task(result)
+        let runnable: Box<dyn Runnable> = serde_json::from_value(task.metadata.clone()).unwrap();
+        let result = self.execute_task(&runnable);
```

Comment:
Deref runnable variable and instead of passing a `&Box<dyn Runnable>` it is better to pass `&dyn Runnable`

Reply:
https://doc.rust-lang.org/std/boxed/struct.Box.html#impl-Deref